azurerm/resource_arm_log_profile_test.go

```diff
@@ -0,0 +1,352 @@
+	account_replication_type = "GRS"
+}
+	
+resource "azurerm_log_profile" "test" {
+	name = "acctestlp-%d"
+
+	categories = [
+		"Action",
+	]
+	
+	locations = [
+		"%s"
+	]
+	
+	storage_account_id  = "${azurerm_storage_account.test.id}"
+
+	retention_policy {
+		enabled = true
+		days = 7
+	}
+}
+`, rInt, location, rString, rInt, location)
+}
+
+func testAccAzureRMLogProfile_servicebusConfig(rInt int, rString string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+	name     = "acctestrg-%d"
+	location = "%s"
+}
+	
+resource "azurerm_servicebus_namespace" "test" {
+  name                = "acctestsbns-%s"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  sku                 = "standard"
+}
+
+resource "azurerm_servicebus_namespace_authorization_rule" "test" {
+  name                = "acctestsbrule-%s"
+  namespace_name      = "${azurerm_servicebus_namespace.test.name}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  listen = true
+  send   = true
+  manage = true
+}
+
+resource "azurerm_log_profile" "test" {
+	name = "acctestlp-%d"
+
+	categories = [
+		"Action"
+	]
+	
+	locations = [
+		"%s",
+	]
+	
+	service_bus_rule_id = "${azurerm_servicebus_namespace_authorization_rule.test.id}"
+	
+	retention_policy {
+		enabled = false
+		days = 0
+	}
+}
+`, rInt, location, rString, rString, rInt, location)
+}
+
+func testAccAzureRMLogProfile_completeConfig(rInt int, rString string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+	name     = "acctestrg-%d"
+	location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "acctestsa%s"
+  resource_group_name      = "${azurerm_resource_group.test.name}"
+  location                 = "${azurerm_resource_group.test.location}"
+  account_tier             = "Standard"
+  account_replication_type = "GRS"
+}
+	
+resource "azurerm_eventhub_namespace" "test" {
+  name                = "acctestehns-%s"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  sku                 = "Standard"
+  capacity            = 2
+}
+
+resource "azurerm_log_profile" "test" {
+	name = "acctestlp-%d"
+
+	categories = [
+		"Action",
+		"Delete",
+		"Write",
+	]
+	
+	locations = [
+		"east asia",
+		"southeastasia",
+		"centralus",
+		"eastus",
+		"eastus2",
+		"westus",
+		"northcentralus",
+		"southcentralus",
+		"northeurope",
+		"westeurope",
+		"japan west",
+		"japaneast",
+		"brazilsouth",
+		"australiaeast",
+		"australiasoutheast",
+		"southindia",
+		"centralindia",
+		"west india",
+		"canadacentral",
+		"canadaeast",
+		"uksouth",
+		"ukwest",
+		"westcentralus",
+		"westus2",
+		"koreacentral",
+		"koreasouth",
+		"francecentral",
+		"francesouth",
+		"australiacentral",
+		"australiacentral2",
```

Comment:
whilst this list passes in Azure Public - this'll fail in Azure China/Germany/Government etc - can we use the `testLocation()` and `testAltLocation()` methods to submit in both of these instead?